Extension/src/Utility/Text/scanner.ts

```diff
@@ -0,0 +1,1018 @@
+        }
+
+        if (escaped) {
+            value = this.unescapeString(value);
+        }
+
+        this.text = text;
+        this.stringValue = value;
+        return this.kind = Kind.StringLiteral;
+    }
+
+    private scanCodeFence() {
+        // skip to the end of this line first
+        const fenceStart = this.scanUntil((ch, chNext) => ch === CharacterCodes.lineFeed || (ch === CharacterCodes.carriageReturn && chNext === CharacterCodes.lineFeed));
+
+        // end when the first column is triple backtick
+        const code = this.scanUntil((ch, chNext, chNextNext) => this.#column === 0 && (ch === CharacterCodes.backtick && chNext === CharacterCodes.backtick && chNextNext === CharacterCodes.backtick));
+
+        this.text = fenceStart + code;
+
+        // code is just the inside contents
+        this.stringValue = code.substring(0, code.length - 3);
+
+        return this.kind = Kind.CodeFence;
+    }
+
+    private unescapeString(text: string) {
+        let result = '';
+        let start = 0;
+        let pos = 0;
+        const end = text.length;
+
+        while (pos < end) {
+            let ch = text.charCodeAt(pos);
+            if (ch !== CharacterCodes.backslash) {
+                pos++;
+                continue;
+            }
+
+            result += text.substring(start, pos);
+            pos++;
+            ch = text.charCodeAt(pos);
+
+            switch (ch) {
+                case CharacterCodes.r:
+                    result += '\r';
+                    break;
+                case CharacterCodes.n:
+                    result += '\n';
+                    break;
+                case CharacterCodes.t:
+                    result += '\t';
+                    break;
+                case CharacterCodes.singleQuote:
+                    result += '\'';
+                    break;
+                case CharacterCodes.doubleQuote:
+                    result += '"';
+                    break;
+                case CharacterCodes.backslash:
+                    result += '\\';
+                    break;
+                case CharacterCodes.backtick:
+                    result += '`';
+                    break;
+                default:
+                    throw new ScannerError('Invalid escape sequence', this.position.line, this.position.column);
+            }
+
+            pos++;
+            start = pos;
+        }
+
+        result += text.substring(start, pos);
+        return result;
+    }
+
+    scanIdentifier() {
+        this.text = this.scanUntil((ch) => !isIdentifierPart(ch));
+        return this.kind = keywords.get(this.text) ?? Kind.Identifier;
+    }
+
+    scanVariable() {
+        this.text = '$';
+        this.text += this.scanUntil((ch) => !isIdentifierPart(ch));
+        return this.kind = Kind.Variable;
+    }
+
+    /**
+   * Returns the zero-based line/column from the given offset
+   * (binary search thru the token start locations)

```

Comment:
Could you find-replace "thru"->"through":

<img width="311" alt="image" src="https://github.com/microsoft/vscode-cpptools/assets/19859882/f4237554-8bfc-4bd6-913f-e96bd6f70486">

Reply:
fixing